File: php/features.php
<?php
declare(strict_types=1);

// AgentGateway SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class AgentGatewayFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new AgentGatewayBaseFeature();
            case "test":
                return new AgentGatewayTestFeature();
            default:
                return new AgentGatewayBaseFeature();
        }
    }
}

File: py/agentgateway_sdk/features.py
# AgentGateway SDK feature factory

from agentgateway_sdk.feature.base_feature import AgentGatewayBaseFeature
from agentgateway_sdk.feature.test_feature import AgentGatewayTestFeature


def _make_feature(name):
    features = {
        "base": lambda: AgentGatewayBaseFeature(),
        "test": lambda: AgentGatewayTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
